Redesign of data flow in frontend

thomaj opened the pull request:
## Changes
The main difference is the addition of Vuex to the project.  This creates a 'store' which contains the query and the response.  This helps the navigation and flow of data through the app.  It makes sure the app doesn't crash due to variables losing their data and being undefined when we use them to display the information retrieved from the API calls.

Also restructured some javascript code that is used to make API calls into individual modules. This makes it easier for code reuse and doesn't clutter the Vue component files.

### Note:
 Will most likely have to run npm install

Commit:
component restructuring

evankozliner committed:
Adds a scraper and json dumper

evankozliner committed:
Adds a list of SOLR commands so we don't need to hunt down the API reference

evankozliner committed:
Added makefile for necessary directories and sqlite db. Short shell script handles scrapping RAR and replacing it

evankozliner committed:
Solr configuration, including schema

Commit:
Finished component restructoring

Commit:
Passing data test

evankozliner committed:
Adds a little flask server that handles SOLR queries

evankozliner committed:
Couple of bug fixes, should be ready for demo

Commit:
Added loading spinner

Commit:
Updated transitions between components

evankozliner committed:
Adds an upload for the writer for immediately after the text file is finished

evankozliner committed:
Updates readme and removes scraper dependency in server

evankozliner committed:
Realized I forgot to upload descriptions, will recreate later. Added published and emotions to response

Commit:
Merge remote-tracking branch 'origin/RAR-addition' into add-RAR-call

Mergining in the server for Retrieve and Rank

evankozliner committed:
Changes port

Commit:
Merge in of RAR-branch and began set call set up

Began set up for api call to the RAR backend server

evankozliner committed:
adds cors for local usage

Commit:
Mergered in fix in the python server

Commit:
Finished adding RAR call to the front end

Articles can be retrieved from the front end and displayed

evankozliner committed:
Adds the description to SOLR

Commit:
Worked on styling

Commit:
Merged in RAR server fix

Commit:
Worked on good article styling

Commit:
Added sentiment category analysis

Added javascript Sentiment module to determine which category the article blongs to and with what confidence.  Also added more styling.

Commit:
Minor fixes

Fixed jump in transition and added headers

Commit:
Actually fixed transition and changes styling

Commit:
Added sidebar component back

Added Sidebar component to the validQuestion component and changed come styling.  Also changed some CSS so the display of the definition could be more dynamic for long definitions

Commit:
Changed method to create

Commit:
Initial addition of vuex

Vuex added to bring state to the application so you can navigate through the webpages and not get error messages and non-renderings due to properties being null.  First addition, so still bugs with the actual data being kept.

Commit:
Finished adding vuex

Finished the general layout of the store and how data is passed around. Also added a few minor updates in some components.

Commit:
Small changes

evankozliner committed:
Fixes issue with short name

Commit:
Pulled out router push from SearchActions initialSearch

Commit:
Added Article component and can now sort by date

Commit:
Articles are now sorted and displayed by date

Articles are sorted by date with the most recent article at the top of the page. The color next to it represents the sentiment towards that article

Commit:
Fixed issue when error occurs during a search where the vue instance is not found and is undefined in Search.vue

Commit:
Merged in commit where articles are sorted by date

evankozliner commented:
Currently getting an error trying to search in this branch. See if you can recreate this issue when you get a moment @thomaj 

Basically the request hangs at the loader

<img width="1430" alt="screen shot 2017-03-22 at 4 52 19 pm" src="https://cloud.githubusercontent.com/assets/4589591/24220314/2d6eefa2-0f20-11e7-9030-789640f5703e.png">

Commit:
Fixed Issue of undefined instance of vue

Fixed the issue where this (which would be the instance of Vue) was undefined in the success side of the method in Search.vue.  This is the same fix for the error side of the method too.
